doc/BGP/BGP-supress-fib-pending.md

```diff
@@ -0,0 +1,804 @@
+  - [6.2. Manifest (if the feature is an Application Extension)](#62-manifest-if-the-feature-is-an-application-extension)
+  - [6.3. CLI/YANG model Enhancements](#63-cliyang-model-enhancements)
+- [7. High-Level Design](#7-high-level-design)
+  - [7.1. FPM plugin migration](#71-fpm-plugin-migration)
+  - [7.2. BGP Docker container startup](#72-bgp-docker-container-startup)
+  - [7.3. RouteOrch](#73-routeorch)
+  - [7.4. FPMsyncd](#74-fpmsyncd)
+  - [7.5. Response Channel Performance considerations](#75-response-channel-performance-considerations)
+  - [7.6. Consistency monitoring script and mitigation action](#76-consistency-monitoring-script-and-mitigation-action)
+- [8. SAI API](#8-sai-api)
+- [9. Warmboot and Fastboot Design Impact](#9-warmboot-and-fastboot-design-impact)
+  - [9.1. Warm Reboot](#91-warm-reboot)
+  - [9.2. Fast Reboot](#92-fast-reboot)
+- [10. Restrictions/Limitations](#10-restrictionslimitations)
+- [11. Testing Requirements/Design](#11-testing-requirementsdesign)
+  - [11.1. Unit Test cases](#111-unit-test-cases)
+  - [11.2. VS Test cases](#112-vs-test-cases)
+  - [11.3. System Test cases](#113-system-test-cases)
+  - [11.4. Performance measurements](#114-performance-measurements)
+- [12. Open/Action items - if any](#12-openaction-items---if-any)
+
+<!-- omit in toc -->
+### 1. Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Sep 15 2022 | Stepan Blyshchak | Initial proposal   |
+
+### 1. Scope
+
+This document describes a feedback mechanism that allows BGP not to advertise routes that haven't been programmed yet.
+
+### 2. Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description                             |
+| ------------------------ | --------------------------------------- |
+| ASIC                     | Application specific integrated circuit |
+| HW                       | Hardware                                |
+| SW                       | Software                                |
+| BGP                      | Border Gateway Protocol                 |
+| FRR                      | Free Range Routing                      |
+| SWSS                     | Switch state service                    |
+| SYNCD                    | ASIC synchronization service            |
+| FPM                      | Forwarding Plane Manager                |
+| SAI                      | Switch Abstraction Interface            |
+| OID                      | Object Identifier                       |
+
+### 3. Overview
+
+As of today, SONiC BGP advertises learnt prefixes regardless of whether these prefixes were successfully programmed into ASIC. While route programming failure is followed by orchagent crash and all services restart, even for successfully created routes there is a short period of time when the peer will be black holing traffic. Also, in the following scenario, a credit loop occurs:
```

Comment:
When there are multiple updates for the given route (there are NH changes) from fpmsyncd to route-orch, route-orch will send the response to every route changes after ASIC programming? or it will send the response to the latest route-update to fpmsyncd? please clarify.

Reply:
Orchagent will send a response for every route update it gets from APPL_DB. As far as I know there is a logic inside ProducerStateTable/ConsumerStateTable that will consolidate multiple SET/DEL operations for a single key which makes orchagent get the most up to date requested field values for a route entry. FRR, on the other hand, currently is only interested in the first route creation status. It does not care about the installation status after the route next hop or other attributes have changed.

If zebra sends RTM_DELROUTE before fpmsyncd sends RTM_NEWROUTE with RTM_F_OFFLOAD, zebra will ignore it, considering it as a stale update.

Reply:
Good to know, thanks for the clarification.